src/universal/components/AvatarInput.tsx

```diff
@@ -0,0 +1,72 @@
+import React, {Component, lazy, Suspense} from 'react'
+import styled from 'react-emotion'
+import RaisedButton from 'universal/components/RaisedButton'
+import StyledError from 'universal/components/StyledError'
+import Confetti from './Confetti'
+
+const HiddenInput = styled('input')({
+  display: 'none'
+})
+
+const Control = styled('div')({
+  display: 'flex',
+  justifyContent: 'center',
+  overflow: 'hidden',
+  padding: '.5rem',
+  position: 'relative'
+})
+
+interface Props {
+  onSubmit: (file: File) => void
+  error?: string
+}
+
+const Confetti = lazy(() =>
+  import(/* webpackChunkName: 'Confetti' */ 'universal/components/Confetti')
+)
+
+class AvatarInput extends Component<Props> {
+  inputRef = React.createRef<HTMLInputElement>()
+  onClick = () => {
+    if (this.inputRef.current) {
+      this.inputRef.current.click()
+    }
+  }
+  onChange = (e: React.ChangeEvent<HTMLInputElement> | React.FormEvent<HTMLFormElement>) => {
+    const {onSubmit} = this.props
+    const {files} = e.currentTarget
+    const imageToUpload = files ? files[0] : null
+    if (!imageToUpload) return
+    // TODO see if we still need the setTimeout
```

Comment:
I dare to ask, what happened without the setTimeout()?

Reply:
actually that's a great question. redux-form needed that, but now that we're vanilla, i think we're OK